cypress-io/cypress
# fix: reduce requests paused and redundant headers

- Closes #26620
- Closes #26622

### Additional details
While investigating possible network slowness within Cypress, I noticed that we might be able to optimize the number of requests that are paused by CDP, electron, and Firefox. This PR's goal is to only pause network requests when necessary in order to associate the AUT.

However, we were also using paused network requests to attempt to associate resource types when a request was made with `fetch` or `xhr`. This was pointed out to me in https://github.com/cypress-io/cypress/pull/25075, meaning we no longer need to associate this in the request via CDP, because it happens in the prerequest logic. This has been removed entirely in the request intercepting logic.

When performance benchmarking, we noticed that when `Fetch.Enabled` was disabled, network blocking time most of the time would see a significant drop. This could be possibly due to other variables, but it seems to be a factor in improving blocking time, especially for certain mimetypes. 

### Steps to test
Unit tests have been adapted, or in some cases removed as they are no longer applicable. The cookie / origin tests serve as a really great regression suite in this case to make sure a) injection is still happening when it needs to and b) cookie logic in the server is still working correctly.

The binary has also been run against our origin providers to make sure no regressions are present that are detectable 

### How has the user experience changed?


### PR Tasks
